scripts/download_data.py

```diff
@@ -0,0 +1,256 @@
+        localIterator = None
+        localVariantFile.close()
+        utils.log('maxPos: {}, minPos: {}'.format(self.maxPos, self.minPos))
+
+    def _processFileName(self, fileName):
+        url = os.path.join(self.getVcfBaseUrl(), fileName)
+        utils.log("Downloading '{}'".format(url))
+        response = urllib2.urlopen(url)
+        megabyte = 1024 * 1024
+        data = response.read(megabyte)
+        lineCountQuota = 1000
+        localFileName, _ = os.path.splitext(fileName)
+        utils.log("Writing '{}'".format(localFileName))
+        with tempfile.NamedTemporaryFile() as binaryFile:
+            binaryFile.write(data)
+            binaryFile.flush()
+            gzipFile = gzip.open(binaryFile.name, "r")
+            outputFile = open(localFileName, "w")
+            lineCount = 0
+            for line in gzipFile:
+                outputFile.write(line)
+                if not line.startswith("#"):
+                    lineCount += 1
+                if lineCount >= lineCountQuota:
+                    break
+            assert lineCount == lineCountQuota
+            outputFile.close()
+            gzipFile.close()
+        utils.log("Compressing '{}'".format(localFileName))
+        utils.runCommand('bgzip -f {}'.format(localFileName))
+        utils.log("Indexing '{}'".format(fileName))
+        utils.runCommand('tabix {}'.format(fileName))
+        self._updatePositions(fileName)
+
+    def downloadVcfs(self):
+        self._prepareDir()
+        fileNames = self._getVcfFilenames()
+        for fileName in fileNames:
+            self._processFileName(fileName)
+        escapeDir()
+
+    def downloadBams(self):
+        dirList = [
+            self.args.dir_name, self.datasetId, 'reads', 'low-coverage']
+        mkdirAndChdirList(dirList)
+        cleanDir()
+        studyMap = {
+            'HG00096': 'GBR',
+            'HG00533': 'CHS',
+            'HG00534': 'CHS',
+        }
+        baseUrl = self.getBamBaseUrl()
+        samples = self.args.samples.split(',')
+        for sample in samples:
+            samplePath = '{}/alignment/'.format(sample)
+            study = studyMap[sample]
+            fileName = (
+                '{}.mapped.ILLUMINA.bwa.{}.'
+                'low_coverage.20120522.bam'.format(sample, study))
+            sampleUrl = os.path.join(baseUrl, samplePath, fileName)
+            utils.log("Downloading index for '{}'".format(sampleUrl))
+            remoteFile = pysam.AlignmentFile(sampleUrl)
+            header = remoteFile.header
+            utils.log("Writing '{}'".format(fileName))
+            localFile = pysam.AlignmentFile(
+                fileName, 'wb', header=header)
+            for reference in remoteFile.references:
+                utils.log("reference {}".format(reference))
+                iterator = remoteFile.fetch(
+                    reference, start=self.minPos, end=self.maxPos)
+                numRecords = 0
+                for record in iterator:
```

Comment:
Using 
```python
for index, record in enumerate(iterator):
     localFile.write(record)
     if index > args.num_reads:
         break
```
would be slightly more compact here. Doesn't make any real difference though, just a suggestion.